libscu-c/src/scu.c

```diff
@@ -351,10 +360,14 @@ _scu_run_test(int idx)
 
 	_scu_after_each();
 
-	_scu_output_test_end(idx, success, asserts,
+	size_t valgrind_errors_after = VALGRIND_COUNT_ERRORS;
+
+	size_t valgrind_error_count = valgrind_errors_after - valgrind_errors_before;
```

Comment:
You mean if VALGRIND_COUNT_ERRORS wraps? 4B valgrind errors is quite bad.

But it turns out `VALGRIND_COUNT_ERRORS` has type "unsigned". So the casting to size_t there isn't really helpful. And if testcase 1 gave UINT_MAX-2 errors, and testcase 2 gave 4 it would give wrong result for the second case (wrapped on 64-bit rather than 32-bit).

It all seems quite theoretical. But I think I'll at least change it to `unsigned` rather than `size_t ` to make it match VALGRIND_COUNT_ERRORS definition.

Reply:
I didn't particularly think about overflow, just if valgrind does something funny. But yeah, it seems far fetched. And if you move it to unsigned, any value should fit into 10 digits, so :+1: